core/src/main/scala/akka/kafka/scaladsl/DiscoverySupport.scala

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright (C) 2014 - 2016 Softwaremill <http://softwaremill.com>
+ * Copyright (C) 2016 - 2019 Lightbend Inc. <http://www.lightbend.com>
+ */
+
+package akka.kafka.scaladsl
+
+import akka.actor.{ActorSystem, ActorSystemImpl}
+import akka.annotation.InternalApi
+import akka.discovery.Discovery
+import akka.kafka.{ConsumerSettings, ProducerSettings}
+import akka.util.JavaDurationConverters._
+import com.typesafe.config.Config
+
+import scala.concurrent.Future
+import scala.concurrent.duration.FiniteDuration
+import scala.util.Failure
+
+/**
+ * Scala API.
+ *
+ * Reads Kafka bootstrap servers from configured sources via [[akka.discovery.Discovery]] configuration.
+ */
+object DiscoverySupport {
+
+  /**
+   * Use Akka Discovery to read the addresses for `serviceName` within `lookupTimeout`.
+   */
+  private def bootstrapServers(
+      serviceName: String,
+      lookupTimeout: FiniteDuration
+  )(implicit system: ActorSystem): Future[String] = {
+    import system.dispatcher
+    val discovery = Discovery(system).discovery
+    discovery.lookup(serviceName, lookupTimeout).map { resolved =>
+      resolved.addresses
+        .map { target =>
+          val port = target.port
+            .getOrElse(throw new IllegalArgumentException(s"port missing for $serviceName ${target.host}"))
+          s"${target.host}:$port"
+        }
+        .mkString(",")
+    }
+  }
+
+  /**
+   * Internal API.
+   *
+   * Expect a `service` section in Config and use Akka Discovery to read the addresses for `name` within `lookup-timeout`.
+   */
+  @InternalApi
+  private[kafka] def bootstrapServers(config: Config)(implicit system: ActorSystem): Future[String] = {
+    checkClassOrThrow(system.asInstanceOf[ActorSystemImpl])
+    if (config.hasPath("service")) {
+      val serviceName = config.getString("service.name")
+      val lookupTimeout = config.getDuration("service.lookup-timeout").asScala
```

Comment:
Maybe for consistency with cluster bootstrap, it should use `service-name` and `resolve-timeout`?

https://github.com/akka/akka-management/blob/d0e1baa3df4985089da7a031fc3c4abfb7013a4b/cluster-bootstrap/src/main/resources/reference.conf#L27-L31
https://github.com/akka/akka-management/blob/d0e1baa3df4985089da7a031fc3c4abfb7013a4b/cluster-bootstrap/src/main/resources/reference.conf#L96-L97

Reply:
Then, if it branched on whether `service-name` is set, a reasonable default `lookup-timeout` could be added to `reference.conf`.

It would also be a nice convenience to support setting `service-name` via a standard environment variable, like cluster bootstrap does with `AKKA_CLUSTER_BOOTSTRAP_SERVICE_NAME`, and Lagom currently does with [`KAFKA_SERVICE_NAME`](https://github.com/lagom/lagom/blob/3298cf630f7efc89575d2ea53cf5f7b7241e14a5/service/core/kafka/client/src/main/resources/reference.conf#L13). In fact, using the same variable as Lagom will make it easier for Lagom users when they upgrade to this version, if they're already setting that from their deployment configuration, as we recommend (e.g., in the [OpenShift deployment guide](https://developer.lightbend.com/guides/openshift-deployment/lagom/preparing-for-production.html#connecting-to-kafka)).